Projects/Contracts/Sources/ContractAction.swift

```diff
@@ -89,3 +89,9 @@ public enum TerminationNavigationAction: ActionProtocol {
     case openTerminationFailScreen
     case openTerminationDeletionScreen
 }
+
+extension ContractAction: Hashable {
+    public func hash(into hasher: inout Hasher) {
+        hasher.combine("\(try! JSONEncoder().encode(self))")
```

Comment:
If you define Hashable directly on the ContractAction and all the associated values are Hashable this shouldn't be necessary, it should do that automatically.

Reply:
We had to do it since it was complaining. It does not make sense to me either.

Reply:
Done!